components/QrCodeOverlay.tsx

```diff
@@ -7,69 +7,103 @@ import {Theme} from './ui/styleUtils';
 import {useTranslation} from 'react-i18next';
 import testIDProps from '../shared/commonUtil';
 import {SvgImage} from './ui/svg';
+// @ts-ignore
+import {generateQRData} from '@mosip/pixelpass';
+import {VerifiableCredential} from '../machines/VerifiableCredential/VCMetaMachine/vc';
+import RNSecureKeyStore, {ACCESSIBLE} from 'react-native-secure-key-store';
+import {DEFAULT_ECL, DEFAULT_QR_HEADER} from '../shared/constants';
+import {VCMetadata} from '../shared/VCMetadata';
 
 export const QrCodeOverlay: React.FC<QrCodeOverlayProps> = props => {
   const {t} = useTranslation('VcDetails');
+  const [qrString, setQrString] = useState('');
 
-  const [isQrOverlayVisible, setIsQrOverlayVisible] = useState(false);
+  async function getQRData(): Promise<string> {
+    let qrData: string;
+    try {
+      qrData = await RNSecureKeyStore.get(props.meta.id);
+    } catch {
+      let qrDataJSON = {
+        verifiableCredential: props.verifiableCredential,
+        meta: props.meta,
+      };
+      qrData = generateQRData(JSON.stringify(qrDataJSON), DEFAULT_QR_HEADER);
+      await RNSecureKeyStore.set(props.meta.id, qrData, {
+        accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+      });
+    }
+    return qrData;
+  }
 
+  useEffect(() => {
+    (async () => {
+      const qrString = await getQRData();
+      setQrString(qrString);
+    })();
+  }, []);
```

Comment:
There is no dependency in useEffect, do we want to reload this every time?
Every change due to useEffect will cause component re-render

Reply:
Updated plz take a look